mtags/src/main/scala-3/scala/meta/internal/pc/PcRenameProvider.scala

```diff
@@ -18,10 +18,24 @@ final class PcRenameProvider(
 ) extends WithSymbolSearchCollector[l.TextEdit](driver, params):
   private val forbiddenMethods =
     Set("equals", "hashCode", "unapply", "unary_!", "!")
+
+  private val soughtSymbolNames = soughtSymbols match
+    case Some((symbols, _)) =>
+      symbols.filterNot(_.isError).map(symbol => symbol.decodedName.toString)
+    case None => Set.empty[String]
+
   def canRenameSymbol(sym: Symbol)(using Context): Boolean =
-    (!sym.is(Method) || !forbiddenMethods(sym.decodedName))
-      && (sym.ownersIterator.drop(1).exists(ow => ow.is(Method))
-        || sym.source.path.isWorksheet)
+    val decodedName = sym.decodedName
+    def isNotForbiddenMethod =
+      !sym.is(Method) || !forbiddenMethods(decodedName)
```

Comment:
```suggestion
    def isForbiddenMethod =
      sym.is(Method) && forbiddenMethods(decodedName)
```
For sake or readability.

Reply:
Fixed